### PR TITLE
[v3]add RPM sample key support

### PR DESCRIPTION
**Pull request by yunguowei**
Base on v2 comments:
1. rephrase git log for implementation explaination.
2. Rename private key and use common varible to define private key name.

### CODE REVIEW

File: recipes-core/meta/signing-keys.bbappend [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+              bb.utils.which(os.getenv('PATH'), "gpg")
+    cmd = '%s --list-keys -a "%s"' % (gpg_bin, keyid)
+    status, output = oe.utils.getstatusoutput(cmd)
+    if status:
+        return False
+    else:
+        return True
+
+python do_import_keys () {
+
```
Comment (jiazhang0) [inline recipes-core/meta/signing-keys.bbappend]: Remove this empty line.

File: recipes-core/meta/signing-keys.bbappend [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            return
+
+        if d.getVar("RPM_GPG_PRIKEY", True):
+            # Import private key of the rpm signing key
+            import_gpg_key(d, d.getVar('RPM_GPG_PRIKEY', True))
+}
+
+addtask do_import_keys before do_export_public_keys
+
+RDEPENDS_${PN} = ""
```
Comment (jiazhang0) [inline recipes-core/meta/signing-keys.bbappend]: You need to add comments here as I said in v2.

Comment (yunguowei) [inline recipes-core/meta/signing-keys.bbappend]: okay, I will move that from git log to here.

File: files/rpm_keys/RPM-GPG-KEY-PULSAR-8 [hunk 3]
```diff
@@ -0,0 +1,31 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
```
Comment (jiazhang0) [inline files/rpm_keys/RPM-GPG-KEY-PULSAR-8]: where is the private key file RPM-GPG-PRIVKEY-Wind-River-Pulsar-8?

In addition, the pubkey file  RPM-GPG-KEY-Wind-River-Pulsar-8 is still necessary for key-store-rpm-key.

Comment (yunguowei) [inline files/rpm_keys/RPM-GPG-KEY-PULSAR-8]: Private key is a bianry, not armor format, so it is not shown in git hub's diff.
Check out git log:

commit 426089bd06df0ef20d45080f1eb1d15a6f691940
Author: Yunguo Wei <yunguo.wei@windriver.com>
Date:   Thu Mar 9 17:45:38 2017 +0800

    sign_rpm: Add sample key support for RPM signing
    
    Also refresh Wind River Pulsar8 Sample Key.
    
    Signed-off-by: Yunguo Wei <yunguo.wei@windriver.com>

 conf/layer.conf                                    |   5 +++++
 files/rpm_keys/RPM-GPG-KEY-PULSAR-8                |  31 +++++++++++++++++++++++++++++++
 files/rpm_keys/RPM-GPG-KEY-pulsar8-sample          |  30 ------------------------------
 files/rpm_keys/RPM-GPG-PRIVKEY-Wind-River-Pulsar-8 | Bin 0 -> 2531 bytes
 files/rpm_keys/RPM-GPG-PRIVKEY-pulsar8-sample      |  59 -----------------------------------------------------------
 templates/default/template.conf                    |   2 ++
 6 files changed, 38 insertions(+), 89 deletions(-)

**Pull request closed without merging**